developer overview updates - new (borked a branch) (#10366)
DataDog/documentation: content/en/developers/_index.md

## Overview

The Developers section contains reference materials for developing on Datadog. You may want to develop on Datadog if there is data you want to see in the product that you are not seeing. If this is the case, Datadog may already support the technology you need. Refer to the table of [commonly requested technologies](#commonly-requested-technologies) to find the product or integration that may fulfill your needs.

If the solution you require is truly unavailable, you can contact [Datadog Support][1] to request a feature. You may also wish to [create your own solution](#creating-your-own-solution) by using the reference materials in this section.

> ### Partners and Datadog Marketplace

Reviewer: ```suggestion
### Partners and the Datadog Marketplace
```

Additionally, you may also be a partner who wants to build on Datadog and contribute to the Datadog Marketplace or to Datadog's community integrations. For more information about becoming a Datadog partner, see the [Partner program page][2].

## Commonly requested technologies

> If there is data you want to monitor with Datadog that you are not seeing, consider the following Datadog products and integrations.

Reviewer: ```suggestion
If there is data you want to monitor with Datadog that you are not seeing, before building something custom, consider the following Datadog products and integrations:
```

| Logs                         | Most applications generate some type of log. Use Datadog's [Log Management][15] for base monitoring. |
| Network traffic and SNMP     | Network devices, such as routers, switches, printers, emit SNMP metrics (Simple Network Management Protocol) which can be collected using [Network Device Monitoring][16]. Additionally, [Network Performance Monitoring][17] tracks all network traffic in and out of a host. |
| OpenMetrics                  | The Agent includes [OpenMetrics and Prometheus checks][18] capable of scraping Prometheus endpoints. |
| Processes                    | To collect resource usage metrics for specific running processes on any host (CPU, memory, I/O, etc.) use the [Processes integration][19]. To query across all your running processes for open-ended debugging use [Live Process Monitoring][20] (like htop without having to SSH).|
| Windows Performance Counters | Windows services or applications that expose performance counters can be collected with the [PDH integration][21]. |

## Creating your own solution

> Developers have several choices for sending unsupported data to Datadog.

Reviewer: ```suggestion
Still not seeing the type of data that you need? Developers have several choices for sending unsupported data to Datadog.
```

- [**DogStatsD**][22] is a metrics aggregation service that accepts custom metrics, events, and service checks.

> - [**Custom checks**][23] enable you to collect metrics from custom applications or systems. [Custom Agent checks][23] are suitable for many basic needs. For more advanced requirements like metrics preprocessing, you may choose to write an [OpenMetrics][24] check. You may also wish to submit a [service][25] check within a custom Agent check.

Reviewer 2: ```suggestion
- [**Custom checks**][23] enable you to collect metrics from custom applications or systems. [Custom Agent checks][23] are suitable for many needs. For more advanced requirements like metrics preprocessing, you may choose to write an [OpenMetrics][24] check. You may also wish to submit a [service][25] check within a custom Agent check.
```

> - [**Integrations**][26] also enable you to collect metrics from custom applications or systems. Integrations are reusable. You can keep your integration private, or you can write a public integration and contribute to Datadog's [repository of community integrations][27] to be used by other developers.

Reviewer 2: ```suggestion
- [**Integrations**][26] also enable you to collect metrics, events, and service checks from custom applications or systems. Integrations are reusable. You may keep your integration private, or  write a public integration contributing to Datadog's [repository of community integrations][27] to be used by other developers.
```

### Should I write a custom check or an integration?

> The primary difference between custom checks and integrations is that integrations are reusable components that can become part of the Datadog's ecosystem. They generally take more effort (time to develop) and are best suited for general use-cases such as application frameworks, open source projects, or commonly used software. For more niche scenarios, such as monitoring services that are not widely used outside your team or organization, writing a custom check may be the most efficient option. However, you may choose to write an integration instead of a custom check if your particular use-case requires you to publish and deploy your solution as a Python wheel (`.whl`).

Reviewer: ```suggestion
The primary difference between custom checks and integrations is that integrations are reusable components that can become part of the Datadog's ecosystem. They generally take more effort (time to develop) and are best suited for general use-cases such as application frameworks, open source projects, or commonly used software. For more niche scenarios, such as monitoring services that are not widely used outside your team or organization, writing a custom check may be the most efficient option. However, you may choose to write an integration instead of a custom check if your particular use-case requires you to publish and deploy your solution as a Python wheel (`.whl`). Custom checks also use custom metrics, which have a cost associated, whereas if an integration gets accepted into the Datadog ecosystem, it is no longer considered a custom metric. For more information about how this might impact cost, see our the [Datadog Pricing Page][29]. 
```

> The primary difference between custom checks and integrations is that integrations are reusable components that can become part of the Datadog's ecosystem. They generally take more effort (time to develop) and are best suited for general use-cases such as application frameworks, open source projects, or commonly used software. For more niche scenarios, such as monitoring services that are not widely used outside your team or organization, writing a custom check may be the most efficient option. However, you may choose to write an integration instead of a custom check if your particular use-case requires you to publish and deploy your solution as a Python wheel (`.whl`). Custom checks also use custom metrics, which have a cost associated, whereas if an integration gets accepted into the Datadog ecosystem, it is no longer considered a custom metric. For more information about how this might impact cost, see our the [Datadog Pricing Page][29]. 

Author: ```suggestion
The primary difference between custom checks and integrations is that integrations are reusable components that can become part of the Datadog's ecosystem. They generally take more effort (time to develop) and are best suited for general use-cases such as application frameworks, open source projects, or commonly used software. For more niche scenarios, such as monitoring services that are not widely used outside your team or organization, writing a custom check may be the most efficient option. However, you may choose to write an integration instead of a custom check if your particular use-case requires you to publish and deploy your solution as a Python wheel (`.whl`). Custom checks also use custom metrics, which have a cost associated, whereas if an integration gets accepted into the Datadog ecosystem, it is no longer considered a custom metric. For more information about how this might impact cost, see the [Datadog Pricing Page][29]. 
```

Reviewer 2: ```suggestion
The primary difference between custom checks and integrations is that integrations are reusable components that can become part of the Datadog's ecosystem. They generally take more effort (time to develop) and are best suited for general use-cases such as application frameworks, open source projects, or commonly used software. For more niche scenarios, such as monitoring services that are not widely used outside your team or organization, writing a custom check may be the most efficient option. However, you may choose to write an integration instead of a custom check if your particular use-case requires you to publish and deploy your solution as a Python wheel (`.whl`). Metrics emitted via Custom checks are considered custom metrics, which have a cost associated, based on your subscription plan. However, once an integration gets accepted into the Datadog ecosystem, metrics that it emits are no longer considered custom metrics, and do not count against your custom metric count. For more information about how this might impact cost, see our the [Datadog Pricing Page][29]. 
```

## Further Reading

{{< partial name="whats-next/whats-next.html" >}}

[1]: /help/
[2]: https://www.datadoghq.com/partner/
[3]: /tracing/
[4]: /amazon_web_services/?tab=allpermissions#overview
[5]: /google_cloud_platform/?tab=datadogussite#overview
[6]: /azure/?tab=azurecliv20#overview
[7]: /alibaba_cloud/#overview
[8]: /integrations/http_check/
[9]: /synthetics/
[10]: /integrations/directory/
[11]: /integrations/java/?tab=host#configuration
[12]: /integrations/solr/
[13]: /integrations/tomcat/
[14]: /integrations/cassandra/
[15]: /logs/
[16]: /network_monitoring/devices/#overview
[17]: /network_monitoring/performance/
[18]: /agent/docker/prometheus
[19]: /integrations/process/
[20]: /graphing/infrastructure/process/?tab=linuxwindows#introduction
[21]: /integrations/pdh_check/
[22]: /developers/dogstatsd/
[23]: /developers/write_agent_check/
[24]: /developers/prometheus/
[25]: /developers/service_checks/
[26]: /developers/integrations/
[27]: https://github.com/DataDog/integrations-extras

> [28]: /developers/marketplace/

Reviewer: ```suggestion
[28]: /developers/marketplace/
[29]: https://www.datadoghq.com/pricing/
```
